### PR TITLE
Doc update async:stream and async:streamsubscription

### PR DESCRIPTION
**Pull request by jpsiitonen**
Stream method examples added
StreamSubscription example added

### CODE REVIEW

File: sdk/lib/async/stream.dart [hunk 1]
```diff
@@ -209,6 +222,17 @@ abstract class Stream<T> {
...
+  /// ```dart
+  /// Future<String> futureTask() async {
+  ///   return await Future.delayed(
+  ///       const Duration(seconds: 5), () => 'Future complete');
+  /// }
+  ///
+  /// final stream = Stream<String>.fromFuture(futureTask());
+  /// stream.listen((event) => print(event),
+  ///     onDone: () => print('Done'), onError: (error) => print(error));
+  /// ```
```
Comment (alexlindroos) [inline sdk/lib/async/stream.dart:240]: Should the output be added here?

File: sdk/lib/async/stream.dart [hunk 2]
```diff
@@ -238,6 +262,22 @@ abstract class Stream<T> {
...
+  ///
+  /// Future<String> doneTask() async {
+  ///   return await Future.delayed(
+  ///       const Duration(seconds: 5), () => 'Future complete');
+  /// }
+  ///
+  /// final stream = Stream<Object>.fromFutures([waitTask(), doneTask()]);
+  /// stream.listen((event) => print(event),
+  ///     onDone: () => print('Done'), onError: (error) => print(error));
+  /// ```
```
Comment (alexlindroos) [inline sdk/lib/async/stream.dart:290]: Should the output be added here?

File: sdk/lib/async/stream.dart [hunk 3]
```diff
@@ -496,6 +551,43 @@ abstract class Stream<T> {
...
   /// then no further events are ever emitted by current subscriptions on
   /// the returned broadcast stream, not even a done event.
+  ///
+  /// Example:
+  /// ```dart
+  /// final stream =
+  ///     Stream<int>.periodic(const Duration(seconds: 1), (count) => count)
+  ///         .take(10);
+  ///
+  /// final broadCastStream = stream.asBroadcastStream(
```
Comment (alexlindroos) [inline sdk/lib/async/stream.dart]: `broadcastStream`

File: sdk/lib/async/stream.dart [hunk 4]
```diff
@@ -496,6 +551,43 @@ abstract class Stream<T> {
...
+  /// final evenStream = broadCastStream.where((event) => event.isEven);
+  /// final evenListener = evenStream.listen((event) {
+  ///   print('Even: $event');
+  /// }, onDone: () => print('Done'));
+  ///
+  /// await Future.delayed(const Duration(seconds: 5));
+  /// // Cancel listeners.
+  /// oddListener.cancel();
+  /// evenListener.cancel();
+  /// ```
```
Comment (alexlindroos) [inline sdk/lib/async/stream.dart:607]: missing output:

Even: 0
Odd: 1
Even: 2
Stream paused: true
Stream cancelled

File: sdk/lib/async/stream.dart [hunk 5]
```diff
@@ -496,6 +551,43 @@ abstract class Stream<T> {
...
+  ///
+  /// final oddStream = broadCastStream.where((event) => event.isOdd);
+  /// final oddListener = oddStream.listen(
+  ///       (event) {
+  ///     print('Odd: $event');
+  ///   },
+  ///   onDone: () => print('Done'),
+  /// );
+  ///
+  /// final evenStream = broadCastStream.where((event) => event.isEven);
```
Comment (alexlindroos) [inline sdk/lib/async/stream.dart]: Even and event is so close to each other so should this be `evenNumberStream`

File: sdk/lib/async/stream.dart [hunk 6]
```diff
@@ -496,6 +551,43 @@ abstract class Stream<T> {
...
+  /// final evenListener = evenStream.listen((event) {
```
Comment (alexlindroos) [inline sdk/lib/async/stream.dart]: Even and event is so close to each other so should this be `evenNumberListener`

File: sdk/lib/async/stream.dart [hunk 7]
```diff
@@ -716,6 +834,16 @@ abstract class Stream<T> {
...
   /// will individually perform the `test` and handle the error.
+  ///
+  /// Example:
+  /// ```dart
+  /// Stream.periodic(const Duration(seconds: 1), (count) {
+  ///   if (count == 2) {
+  ///     throw Exception('custom error');
+  ///   }
+  ///   return count;
+  /// }).take(10).handleError((error) => print(error)).listen(print);
```
Comment (alexlindroos) [inline sdk/lib/async/stream.dart]: Missing the output :

0
1
Exception: custom error
3
4
5
6
7
8
9
10

**Review by stepthomp — COMMENTED** [review]
The following sentences feel incomplete (or at least otherwise don't make sense to me):
“The resulting stream is a broadcast stream if [source] is.”
“The returned stream is a broadcast stream if this stream is.” (This occurs five times in the text.)

**Pull request closed without merging**